ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/user.h

```diff
@@ -664,7 +666,19 @@ typedef signed int _SlFd_t;
     \warning
 */
 #ifndef SL_INC_INTERNAL_ERRNO
-#define slcb_SetErrno
+/*
+ * Zephyr Port: use Zephyr SDK's errno.h definitions, and supply those missing
+ * to allow the SimpleLink driver.c to compile
+ * Also, supply the external errno setter function.
+ */
+#include <errno.h>
```

Comment:
Actually, user.h is one of the files required to change for porting SimpleLink to another OS.  That's why it's (in latest SDK releases) placed in the wifi/**porting/** directory.

The goal is to avoid changing another imported file (driver.c), which needs to include the OS's BSD errno definition when SimpleLink's internal errno mechanism is not used.   Those changes are allowed to be made in the user.h header file (in the wifi/porting/ directory), which gets included in simplelink.h, which driver.c includes.

From driver.c:
#ifdef SL_INC_INTERNAL_ERRNO
#include <ti/drivers/net/wifi/bsd/errno.h>
#endif

So, one alternative would be to modify driver.c directly to include Zephyr's errno.h, but that is not allowed.  I'm not sure there is a different alternative.

I think this change is in the proper file.